lstchain/reco/r0_to_dl1.py

```diff
@@ -274,6 +274,10 @@ def r0_to_dl1(
     event_iter = iter(source)
     first_event = next(event_iter)
 
+    for telescope_id in first_event.r0.tels_with_data:
```

Comment:
I think this nicely solves the FIX ME that was included there, but maybe @moralejo can comment since he is the one that added it.

Reply:
What happens if it is not there? foclen and mirror area will just be undefined?

Reply:
Do you mean in the `first_event`?

Reply:
do you mean the first event?

Reply:
(8 seconds difference in the response 😄)

Reply:
Sorry, read it as `if ... in` not `for ... in`

Reply:
Actually, that fixme does not really apply, as it is not reading anything, just looking it up in a datastructure, which is **very** fast.

Reply:
In that case, should I change it back?

Reply:
you can leave it, just wanted to comment that it's not a performance problem.

Reply:
> you can leave it, just wanted to comment that it's not a performance problem.

The new solution is I think valid only for one telescope - and that is what we have, so no worries. But in principle we should create two arrays foclen, and mirror_area, with per-telescope values.

Reply:
Why @moralejo? It's already in a nice datastructure that is fast to access. Why create other arrays?

Reply:
The old solution *was not a performance issue*. Please just keep it.

Reply:
Should I modify back to defining `foclen` and `mirror_area` inside the event loop then?

Reply:
But it does not really matter, as the script needs to be changed anyway for the next ctapipe release and we won't have multiple telescopes until then